### ISSUE TITLE
Support for React 18

### ISSUE THREAD

**Issue by cabutler10** [google-map-react/google-map-react#1129]
**Describe the bug 🐛**

I recently upgraded to React 18 and now my map no longer works. I saw online that the latest version of google-map-react should include support for React 18, however I do not see any documentation if there is some gotcha or breaking change that I would need to do for correct support within this library? I tired to run your demo in a sandbox (see below link) and see that same error that I am experiencing in my project. 

**To Reproduce 🔍**

Please see this code sandbox of your example using React 18;
https://codesandbox.io/s/boring-cori-wk1si7?file=/src/App.js:55-71

**Expected behavior 💭**

Example demo from docs should work with the latest version of React.

**Comment by Trkzi-Omar** [comment]
Same issue here, the demo from the docs doesn't seem to work with React 18.

**Comment by isaacrowntree** [comment]
Can confirm is broken with React 18. There are three errors:

TypeError: Cannot read properties of undefined (reading 'emit')
TypeError: Cannot read properties of undefined (reading 'dispose')
TypeError: Cannot read properties of undefined (reading 'removeListener')

These are related to `componentDidUpdate` and `componentWillUnmount`

**Comment by hanorine** [comment]
I am having the same issue, console errors with:

```
Uncaught TypeError: Cannot read properties of null (reading 'props')
    at o.getChildren (marker_dispatcher.js:10:1)
    at o._getState (google_map_markers.js:100:1)
    at new o (google_map_markers.js:248:1)
    at constructClassInstance (react-dom.development.js:14323:1)
    at updateClassComponent (react-dom.development.js:19688:1)
    at beginWork (react-dom.development.js:21611:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at HTMLUnknownElement.nrWrapper ((index):1519:39)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)

(index):1521 Uncaught TypeError: Cannot read properties of null (reading 'props')
    at o.getChildren (marker_dispatcher.js:10:1)
    at o._getState (google_map_markers.js:100:1)
    at new o (google_map_markers.js:248:1)
    at constructClassInstance (react-dom.development.js:14323:1)
    at updateClassComponent (react-dom.development.js:19688:1)
    at beginWork (react-dom.development.js:21611:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at HTMLUnknownElement.nrWrapper ((index):1519:39)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)

The above error occurred in the <o> component:

    at o (https://localhost:44316/static/js/bundle.js:126481:19)
    at div
    at o (https://localhost:44316/static/js/bundle.js:127054:16)
    at GoogleMap (https://localhost:44316/static/js/bundle.js:39537:5)
    at div
    at https://localhost:44316/static/js/bundle.js:169890:8
    at div
    at https://localhost:44316/static/js/bundle.js:172979:5
    at div
    at https://localhost:44316/static/js/bundle.js:170046:5
    at SearchResults (https://localhost:44316/static/js/bundle.js:52538:122)
    at Routes (https://localhost:44316/static/js/bundle.js:218495:5)
    at QualityConnectProvider (https://localhost:44316/static/js/bundle.js:41536:5)
    at QualityConnectProviderAndRouting (https://localhost:44316/static/js/bundle.js:41170:76)
    at Routes (https://localhost:44316/static/js/bundle.js:218495:5)
    at ErrorBoundary (https://localhost:44316/static/js/bundle.js:60963:5)
    at I18nextWithTranslation (https://localhost:44316/static/js/bundle.js:206950:31)
    at div
    at div
    at Layout (https://localhost:44316/static/js/bundle.js:7706:21)
    at App (https://localhost:44316/static/js/bundle.js:7245:84)
    at ActiveModalsProvider (https://localhost:44316/static/js/bundle.js:10119:5)
    at AppRoutesAndNavigationProvider (https://localhost:44316/static/js/bundle.js:7573:5)
    at BuildSession (https://localhost:44316/static/js/bundle.js:83760:69)
    at I18nextProvider (https://localhost:44316/static/js/bundle.js:206066:19)
    at Router (https://localhost:44316/static/js/bundle.js:218428:15)
    at HashRouter (https://localhost:44316/static/js/bundle.js:217271:5)
    at Provider (https://localhost:44316/static/js/bundle.js:214554:5)
    at FlagsProvider (https://localhost:44316/static/js/bundle.js:123599:28)

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.
```

**Comment by hanorine** [comment]
Anyone else come across this issue and lack of maintainers here, I would suggest you look at  [react-google-maps/api](https://github.com/JustFly1984/react-google-maps-api) as an alternative.

**Comment by itsmichaeldiego** [comment]
This should be fixed by now! 

Relates to #1102

**Comment by pikilon** [comment]
> This should be fixed by now!
> 
> Relates to #1102

I have installed latest (2.2.0) and it is not working if you have `StrictMode` wrapper in the React 18 app.
these are the main errors I see in de console
![image](https://user-images.githubusercontent.com/3034213/188837354-0246bcc4-1b20-4362-bf78-adf98622ab7b.png)

**Comment by DercilioFontes** [comment]
In development, it only works for React 18 when removing the strict mode (`<React.StrictMode>`). That is not ideal. But it works when built or if you remove/comment out the strict mode.

**Comment by nikita-ks** [comment]
the same issue

**Comment by OmarSalah1911** [comment]
In development, it only works for React 18 when removing the strict mode (<React.StrictMode>).

**Comment by amabirbd** [comment]
With Nextjs this problem is solved  by setting restrict mode to false in the Nextjs config file.

**Comment by yakushstanislav** [comment]
the same issue

**Comment by ifightcrime** [comment]
My comment on this other issue might help: https://github.com/google-map-react/google-map-react/issues/1110#issuecomment-1298946530

TL;DR I just dropped this library in favor of using the Google library and `@googlemaps/react-wrapper` directly.

**Comment by giorgiabosello** [comment]
I've made a repository myself which is working with React 18. I've abandoned this one, since it seems it is not maintained anymore.
[LINK](https://github.com/giorgiabosello/google-maps-react-markers)

**Comment by dpalaniichuk** [comment]
confirm , when removed <React.StrictMode> it works

**Comment by whiteblackswan** [comment]
After upgrading to react@18 my google maps crashing on render.
BootstrapURL Keys doesn't work when I using it with Google Map React. The following error occurs when passing a Marker component to Google Map React with children. Please let me know if you know how to solve this problem.

`Uncaught TypeError: Cannot read properties of undefined (reading 'dispose')
    at r.componentWillUnmount (google_map.js:455:1)
    at callComponentWillUnmountWithTimer (react-dom.development.js:22852:1)
    at safelyCallComponentWillUnmount (react-dom.development.js:22873:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24129:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24018:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24118:1)
    at recursivelyTraverseDeletionEffects (react-dom.development.js:23989:1)
    at commitDeletionEffectsOnFiber (react-dom.development.js:24018:1)`

**Comment by nw-kohei-ito** [comment]
I would like to use with strictmode. Please reopen this issue.

**Comment by Demetri0** [comment]
Who in their right mind would turn StrictMode off?

**Comment by nitesh047** [comment]
Has anyone solved this issue??
